src/amo/components/AddonReview.js

```diff
@@ -46,20 +48,24 @@ export class AddonReviewBase extends React.Component {
     const { reviewBody, reviewTitle } = this.state;
     event.preventDefault();
     event.stopPropagation();
+
+    const addonSlug = this.props.review.addonSlug;
+    const apiState = this.props.apiState;
+
     const params = {
+      addonSlug,
+      apiState,
       body: reviewBody,
       title: reviewTitle,
-      addonSlug: this.props.review.addonSlug,
       errorHandler: this.props.errorHandler,
       reviewId: this.props.review.id,
-      apiState: this.props.apiState,
     };
     // TODO: render a progress indicator in the UI.
     // https://github.com/mozilla/addons-frontend/issues/1156
     return this.props.updateReviewText(params)
-      .then(() => {
-        overlayCard.hide();
-      });
+      // This will update the review count on the add-on detail page.
+      .then(() => this.props.refreshAddon({ addonSlug, apiState }))
```

Comment:
I actually think that's okay UI; usually the problem with showing an empty/skeleton page then loading in content after the API request is done is content jumping around the page. In this case it's just the review count number which shouldn't case content to jump. I say close it and run the API request in the background if it's not a PITA.